plugin.audio.ampache/resources/lib/ampache_connect.py

```diff
@@ -0,0 +1,254 @@
+                xbmc.log("AmpachePlugin::handle_request: ssl",xbmc.LOGDEBUG)
+            else:
+                response = urllib.request.urlopen(req, timeout=400)
+                xbmc.log("AmpachePlugin::handle_request: nossl",xbmc.LOGDEBUG)
+        except urllib.error.HTTPError as e:
+            xbmc.log("AmpachePlugin::handle_request: HTTPError " +\
+                    repr(e),xbmc.LOGDEBUG)
+            xbmc.executebuiltin("ConnectionError" )
+            raise self.ConnectionError
+        except urllib.error.URLError as e:
+            xbmc.log("AmpachePlugin::handle_request: URLError " +\
+                    repr(e),xbmc.LOGDEBUG)
+            xbmc.executebuiltin("ConnectionError" )
+            raise self.ConnectionError
+        except Exception  as e:
+            xbmc.log("AmpachePlugin::handle_request: Generic Error "  +\
+                    repr(e),xbmc.LOGDEBUG)
+            xbmc.executebuiltin("ConnectionError" )
+            raise self.ConnectionError
+        headers = response.headers
+        contents = response.read()
+        response.close()
+        return headers,contents
+
+    def AMPACHECONNECT(self,showok=False):
+        amp_notif = ""
+        version = 350001
+        socket.setdefaulttimeout(3600)
+        nTime = int(time.time())
+        use_api_key = self._connectionData["use_api_key"]
+        if ut.strBool_to_bool(use_api_key):
+            xbmc.log("AmpachePlugin::AMPACHECONNECT api_key",xbmc.LOGDEBUG)
+            myURL = self.get_auth_key_login_url()
+        else: 
+            xbmc.log("AmpachePlugin::AMPACHECONNECT login password",xbmc.LOGDEBUG)
+            myURL = self.get_user_pwd_login_url(nTime)
+        try:
+            headers,contents = self.handle_request(myURL)
+        except self.ConnectionError:
+            xbmc.log("AmpachePlugin::AMPACHECONNECT ConnectionError",xbmc.LOGDEBUG)
+            amp_notif = "Notification(" + ut.tString(30198)  +  "," +\
+                    ut.tString(30202)   + ")"
+            #connection error
+            xbmc.executebuiltin(amp_notif)
+            raise self.ConnectionError
+        except Exception  as e:
+            xbmc.log("AmpachePlugin::AMPACHECONNECT: Generic Error "  +\
+                    repr(e),xbmc.LOGDEBUG)
+        xbmc.log("AmpachePlugin::AMPACHECONNECT ConnectionOk",xbmc.LOGDEBUG)
+        try:
+            xbmc.log("AmpachePlugin::AMPACHECONNECT: contents " +\
+                    contents.decode(),xbmc.LOGDEBUG)
+        except Exception as e:
+            xbmc.log("AmpachePlugin::AMPACHECONNECT: unable to print contents " + \
+                   repr(e) , xbmc.LOGDEBUG)
+        tree=ET.XML(contents)
+        errormess = tree.findtext('error')
+        if errormess:
+            errornode = tree.find("error")
+            if errornode.attrib["code"]=="401":
+                if "time" in errormess:
+                    amp_notif = "Notification(" + ut.tString(30198)   +\
```

Comment:
please replace all entries of `xbmc.executebuiltin(amp_notif)` by the python api equivalent: `xbmcgui.Dialog().notification()`

Reply:
Ok, no problem

Reply:
done